distribution/src/bin/elasticsearch-env

```diff
@@ -126,7 +126,7 @@ if [[ "$ES_DISTRIBUTION_TYPE" == "docker" ]]; then
       es_opt="-E${envvar_key}=${envvar_value}"
       es_arg_array+=("${es_opt}")
     fi
-  done <<< "$(env | grep -E '^[a-z0-9_]+(\.[a-z0-9_]+)+=')"
+  done <<< "$(env | grep -E '^[a-z0-9_]+(.[a-z0-9_]+)+=')"
```

Comment:
Why does removing the backslash fix the issue with the oidc tests? Are we not simply matching any character with `.` now?

Reply:
Quite right. I was confused with the literal interpretation of dots within _square_ brackets. I undid that change. Instead, I had to update the regex to allow for _dashes_ in the variable name. That worked for my local test script and with the gradle task you provided.

Reply:
Nice, let's see what CI says then.

Reply:
I pushed the changes, but this change would make the regex quite a bit more strict. Before the variable would just have to _start_ with the character classes as specified and it didn't care what came after it. I am not sure if that was intended, but making the regex more strict would have impact on users with variable names containing any characters beyond the ```[-a-z0-9_]``` limitation as pushed in my last commit.